client/extensions/woocommerce/woocommerce-services/state/packages/actions.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Internal dependencies
+ */
+import * as api from '../../api';
+import {
+	WOOCOMMERCE_SERVICES_PACKAGES_ADD_PACKAGE,
+	WOOCOMMERCE_SERVICES_PACKAGES_REMOVE_PACKAGE,
+	WOOCOMMERCE_SERVICES_PACKAGES_EDIT_PACKAGE,
+	WOOCOMMERCE_SERVICES_PACKAGES_DISMISS_MODAL,
+	WOOCOMMERCE_SERVICES_PACKAGES_SET_IS_SAVING,
+	WOOCOMMERCE_SERVICES_PACKAGES_SET_MODAL_ERRORS,
+	WOOCOMMERCE_SERVICES_PACKAGES_SET_SELECTED_PRESET,
+	WOOCOMMERCE_SERVICES_PACKAGES_UPDATE_PACKAGES_FIELD,
+	WOOCOMMERCE_SERVICES_PACKAGES_SAVE_PACKAGE,
+	WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_OUTER_DIMENSIONS,
+	WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_ALL,
+	WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_PACKAGE,
+	WOOCOMMERCE_SERVICES_PACKAGES_SET_IS_FETCHING,
+	WOOCOMMERCE_SERVICES_PACKAGES_INIT_PACKAGES_FORM,
+} from '../action-types';
+import { getPackagesForm } from './selectors';
+
+export const addPackage = ( siteId ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_ADD_PACKAGE,
+	siteId,
+} );
+
+export const removePackage = ( siteId, index ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_REMOVE_PACKAGE,
+	index,
+	siteId,
+} );
+
+export const editPackage = ( siteId, packageToEdit ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_EDIT_PACKAGE,
+	'package': packageToEdit,
+	siteId,
+} );
+
+export const dismissModal = ( siteId ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_DISMISS_MODAL,
+	siteId,
+} );
+
+export const setSelectedPreset = ( siteId, value ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_SET_SELECTED_PRESET,
+	value,
+	siteId,
+} );
+
+export const savePackage = ( siteId, packageData ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_SAVE_PACKAGE,
+	packageData,
+	siteId,
+} );
+
+export const updatePackagesField = ( siteId, newValues ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_UPDATE_PACKAGES_FIELD,
+	values: newValues,
+	siteId,
+} );
+
+export const toggleOuterDimensions = ( siteId ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_OUTER_DIMENSIONS,
+	siteId,
+} );
+
+export const toggleAll = ( siteId, serviceId, groupId, checked ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_ALL,
+	serviceId,
+	groupId,
+	checked,
+	siteId,
+} );
+
+export const togglePackage = ( siteId, serviceId, packageId ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_PACKAGE,
+	serviceId,
+	packageId,
+	siteId,
+} );
+
+export const setModalErrors = ( siteId, value ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_SET_MODAL_ERRORS,
+	value,
+	siteId,
+} );
+
+export const setIsSaving = ( siteId, isSaving ) => ( {
+	type: WOOCOMMERCE_SERVICES_PACKAGES_SET_IS_SAVING,
+	isSaving,
+	siteId,
+} );
+
+export const fetchSettings = ( siteId ) => ( dispatch, getState ) => {
+	const form = getPackagesForm( getState(), siteId );
+
+	if ( form && ( form.packages || form.isFetching ) ) {
+		return;
+	}
+	dispatch( { type: WOOCOMMERCE_SERVICES_PACKAGES_SET_IS_FETCHING, isFetching: true, siteId } );
```

Comment:
Is it possible to get this far with a null/undefined siteId? If so, should test for that before dispatching and api-getting